modules/ui/tools/3dmap/3d_map.js

```diff
@@ -0,0 +1,199 @@
+import { select as d3_select } from 'd3-selection';
+
+import { Map } from '../../../3drenderer/Map';
+
+import { t } from '../../../core/localizer';
+import { uiCmd } from '../../cmd';
+
+/*
+ * ui3DMap is a ui panel containing a maplibre 3D Map for visualizing buildings, roads, and areas.
+ * @param {*} context
+ * @returns
+ */
+export function ui3DMap(context) {
+  function threeDMap(selection) {
+    let wrap = d3_select(null);
+    let _isHidden = true; // start out hidden
+    let _map;
+
+    function redraw() {
+      if (_isHidden) return;
+      updateProjection();
+      featuresToGeoJSON();
+    }
+
+    function updateProjection() {
+      // Since the bounds are intended to wrap a box around a perfectly orthogonal view,
+      // for a pitched, isometric view we need to enlarge the box a bit to display more buildings.
+      let extent = context.map().extent();
+      let center = extent.center();
+      extent.padByMeters(100);
+      let bounds = [extent.min, extent.max];
+
+
+      _map.map.jumpTo({ center: center , zoom: context.map().zoom()-3});
+      // _map.map.fitBounds((this.bounds = bounds)), {duration: 0};
+    }
+
+    function featuresToGeoJSON() {
+      var mainmap = context.map();
+      const entities = context.history().intersects(mainmap.extent());
+      const buildingEnts = entities.filter((ent) => {
+        const tags = Object.keys(ent.tags).filter((tagname) =>
+          tagname.startsWith('building')
+        );
+        return tags.length > 0;
+      });
+      const highwayEnts = entities.filter((ent) => {
+        const tags = Object.keys(ent.tags).filter((tagname) =>
+          tagname.startsWith('highway')
+        );
+        return tags.length > 0;
+      });
+      const areaEnts = entities.filter((ent) => {
+        const tags = Object.keys(ent.tags).filter((tagname) =>
+          tagname.startsWith('highway')
+        );
+        return tags.length > 0;
+      });
+      generateRoadLayer(context, areaEnts, _map);
```

Comment:
Looks like I need to fix up some naming here- I started the area entity work but didn't quite get to plumbing it fully.

Reply:
Fixed! Areas are now added and these names make sense.